classes/device.js

```diff
@@ -1,11 +1,10 @@
 const _ = require('lodash');
 const events = require('events');
-const Validator = require('../lib/validator.js');
 const Datastore = require('../lib/datastore.js');
 const Logger = require('../lib/logger.js');
 const { safeLogString, noop, noopPromise } = require('../lib/utils.js');
 
-module.exports = function deviceBaseClass(HomeNode, deviceConfig, instanceConfig) {
+module.exports = function deviceClassClass(HomeNode, deviceConfig, instanceConfig) {
```

Comment:
Yes, because it's being called with the `new` keyword it must be a full function. Also the named function helps in stack traces. I agree a rework to ES6 could be cool at some point if there are benefits.